Reference/Scheduling/index-v9.md

```diff
@@ -0,0 +1,184 @@
+                }
+            }
+
+            // Remember to complete the scope when done.
+            scope.Complete();
+            return Task.CompletedTask;
+        }
+    }
+}
+
+```
+
+### Registering with extension method
+
+First we need to create our extension method where we register the hosted service with `AddHostedService`:
+
+```C#
+using Microsoft.Extensions.DependencyInjection;
+using Umbraco.Cms.Core.DependencyInjection;
+
+namespace Umbraco.Cms.Web.UI
+{
+    public static class BuilderExtensions
+    {
+        public static IUmbracoBuilder AddHostedServices(this IUmbracoBuilder builder)
+        {
+            builder.Services.AddHostedService<CleanUpYourRoom>();
+            return builder;
+        }
+    }
+}
+```
+
+Now we can invoke it in the `ConfigureServices` method in `Startup.cs`:
+
+```C#
+public void ConfigureServices(IServiceCollection services)
+{
+#pragma warning disable IDE0022 // Use expression body for methods
+    services.AddUmbraco(_env, _config)
+        .AddBackOffice()
+        .AddWebsite()
+        .AddComposers()
+        .AddHostedServices() // Register CleanUpYourRoom
+        .Build();
+#pragma warning restore IDE0022 // Use expression body for methods
+}
+```
+
+### Registering with a composer
+
+All we need to do here is to create the composer where we register the hosted service with `AddHostedService`, which will be run automatically:
+
+```C#
+using Microsoft.Extensions.DependencyInjection;
+using Umbraco.Cms.Core.Composing;
+using Umbraco.Cms.Core.DependencyInjection;
+
+namespace Umbraco.Cms.Web.UI
+{
+    public class CleanUpYourRoomComposer : IComposer
+    {
+        public void Compose(IUmbracoBuilder builder)
+        {
+            builder.Services.AddHostedService<CleanUpYourRoom>();
+        }
+    }
+}
+
+```
+
+## RecurringHostedServiceBase
+
+This class provides the base class for any hosted service. 
+
+You can override the `PerformExecuteAsync` method to implement the class. Hosted services are always run asynchronously.
+
+The `RecurringHostedServiceBase` is a base class that implements the netcore interface `IHostedService`, and makes the task recurring, if you don't need your task to run recurringly you can implement `IHostedService` yourself, and register your hosted service in the same way. For more information about hosted services, take a look at the [Microsoft documentation](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-5.0).
+
+## BackgroundTaskRunner Notifications
+
+In earlier versions of Umbraco, there were a series of events triggered by background tasks, with the switch to notifications this no longer exists, however, fear not, because you can publish any custom notification you desire from within you background task. For more information about creating and publishing your own custon notifications see: [Creating and Publishing Custom Notifications](../Notifications/Creating-And-Publishing-Notifications.md)
```

Comment:
```suggestion
In earlier versions of Umbraco, there were a series of events triggered by background tasks, with the switch to notifications this no longer exists, however, fear not, because you can publish any custom notification you desire from within your background task. For more information about creating and publishing your own custom notifications see: [Creating and Publishing Custom Notifications](../Notifications/Creating-And-Publishing-Notifications.md)
```